e2e/scripts/st_arrow_line_chart_add_rows_special.py

```diff
@@ -37,4 +37,4 @@
     df2 = pd.DataFrame(data=[[current_time, count]], columns=["t", "y"]).set_index("t")
     # This issue is Arrow specific, that's why st._legacy_add_rows doesn't have a similar test.
     line_chart._arrow_add_rows(df2)
-    time.sleep(1)
+    time.sleep(0.1)
```

Comment:
(Is this intended to be part of the PR?)

Reply:
Whoops -- I initially tried reducing this time to see if it reduced flakiness (but eventually ended up solving a bunch of issues with flaky tests that merging this PR would otherwise uncover in #4276)

At this point, this can be reverted to what it was